File: src/TP3/Ej9/Ellipse.java
package TP3.Ej9;

public class Ellipse extends Figure{
    protected Point center;
    protected double sMayorAxis,sMinorAxis;
    Ellipse(Point center,double sMayorAxis, double sMinorAxis){
        this.center=center;
        this.sMayorAxis=sMayorAxis;
        this.sMinorAxis=sMinorAxis;
    }

    @Override
    public double area() {
        return Math.PI / 4 * sMayorAxis * sMinorAxis;
    }

    @Override
    public double perimeter() {
        return Math.PI / 2 * (sMayorAxis + sMinorAxis);
    }

    @Override
    public String toString() {
        return "Elipse [Centro: {%.2f , %.2f}, DMayor: %.2f, DMenor: %.2f]".formatted(center.getX(),center.getY(),sMayorAxis,sMinorAxis);
    }
}

File: Ruby/Guias/G7/Ej4/ellipse.rb
require_relative 'figure'
class Ellipse < Figure
  def initialize(inicio,ancho,alto)
    raise "Se ingresaron datos de clases invalidas"if !inicio.is_a?(Point) && (!ancho.is_a?(Integer) || !ancho.is_a?(Float)) && (!alto.is_a?(Integer) || !alto.is_a?(Float))
    @inicio = inicio
    @ancho = ancho
    @alto = alto
  end

  def perimeter
    2*Math::PI*((@ancho**2+@alto**2)/2)**0.5
  end

  def area
    Math::PI*@alto*@ancho
  end
  def to_s
    "Ellipse: [ Centro: #{@inicio} , DMayor: #{@ancho} , DMenor: #{@alto} ]"
  end
end